addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/OpenApiSpider.java

```diff
@@ -64,11 +64,15 @@ public boolean parseResource(HttpMessage message, Source source, int depth) {
     @Override
     public boolean canParseResource(HttpMessage message, String path, boolean wasAlreadyConsumed) {
         try {
-            String contentType = message.getResponseHeader().getHeader(HttpHeader.CONTENT_TYPE);
-            if (contentType.toLowerCase().startsWith("text")
-                    && message.getResponseBody().toString().toLowerCase().startsWith("swagger:")) {
+            String contentType =
+                    message.getResponseHeader().getHeader(HttpHeader.CONTENT_TYPE).toLowerCase();
+            String responseBodyStart =
+                    message.getResponseBody().toString().substring(0, 8).toLowerCase();
```

Comment:
String.substring will throw an exception if the body length is less than 8, so should check for that first.

Reply:
Of course! How silly of me to miss that.